File: base/src/main/java/com/iamrajendra/base/utils/RxBus.java
package com.iamrajendra.base.utils;

import com.iamrajendra.base.data.Rout;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

public class RxBus<T> {

    public RxBus() {
    }

    private PublishSubject<T> bus = PublishSubject.create();

    public void send(T o) {
        bus.onNext(o);
    }

    public Observable<T> toObservable() {
        return bus;
    }

}

File: base/src/main/java/com/iamrajendra/base/utils/RxBus.kt
package com.iamrajendra.base.utils

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

class RxBus<T> {
    private val bus: PublishSubject<T> = PublishSubject.create()
    fun send(o: T) {
        bus.onNext(o)
    }

    fun toObservable(): Observable<T> {
        return bus
    }
}